src/libsyntax/parse/parser.rs

```diff
@@ -233,6 +233,57 @@ pub struct ModulePathError {
     pub help_msg: String,
 }
 
+pub enum Errors {
+    FileNotFoundForModule { mod_name: String,
+                            default_path: String,
+                            secondary_path: String,
+                            dir_path: String },
+    DuplicatePaths { mod_name: String, default_path: String, secondary_path: String },
+    UselessDocComment,
+    InclusiveRangeWithNoEnd,
```

Comment:
Do we need this variant? I think it would be simpler to just emit the error directly in the relevant parsing function.